File: TpServer/src/entity/weekpunch.ts
import { Table, Column, PrimaryGeneratedColumn, OneToMany, Index, VersionColumn } from "typeorm";
import { Punch } from './punch';
@Table()
export class WeekPunch {
    @PrimaryGeneratedColumn()
    id: number;
    @Column()
    @Index()
    week: number;
    @OneToMany(type => Punch, punch => punch.week) // note: we will create author property in the Punch class below
    punches: Punch[];
    @VersionColumn()
    version: number;
}

File: TpDotNetCore/Domain/Punches/WeekPunch.cs
using System.Collections.Generic;

namespace TpDotNetCore.Domain.Punches
{
    public class WeekPunch
    {
        public int Id { get; set; }
        
        public int Week { get; set; }
        
        public List<Punch> Punches { get; set; }
    }
}